springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/beans/DefaultBeanMethodsScanner.java

```diff
@@ -0,0 +1,36 @@
+package io.github.stavshamir.springwolf.asyncapi.scanners.beans;
+
+import io.github.stavshamir.springwolf.asyncapi.scanners.classes.ConfigurationClassScanner;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Service;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toSet;
+
+@Service
+@RequiredArgsConstructor
+public class DefaultBeanMethodsScanner implements BeanMethodsScanner {
+
+    private final ConfigurationClassScanner configurationClassScanner;
+
+    @Override
+    public Set<Method> getBeanMethods() {
+        Set<Class<?>> configurationClasses = configurationClassScanner.scan();
+
+        Stream<Method> methods = configurationClasses.stream()
+                .map(Class::getDeclaredMethods)
+                .map(Arrays::asList)
+                .flatMap(List::stream);
+
+        return methods
+                .filter(method -> method.isAnnotationPresent(Bean.class))
+                .collect(toSet());
```

Comment:
There's a `TreeSet`, but I am wondering if sorting should be at the scanner level or at the serializer level. If we do the sorting on the serializer we only have to do it once at one place, instead of doing it in every plugin.

Reply:
Good idea to do the sorting only once, didnt notice the explicit two steps yet. My primary motivation is deterministic integration tests, there do it in the serializer would be fully sufficient - and probably easier and at the right level.